### ISSUE TITLE
TT-8981 Pump crashes on non UTF-8 URLs

### ISSUE THREAD

**Issue by patriziobrunops** [TykTechnologies/tyk-pump#644]
During a vulnerability scan, we noticed `pump` crashing for not being able to handle URLs containing non-UTF8 characters
```


2023-05-22 06:18:00 | panic: label value "/\xc0\xae\xc0\xae/\xc0\xae\xc0\xae/\xc0\xae\xc0\xae/etc/passwd" is not valid UTF-8 |  
-- | -- | --
  |   | 2023-05-22 06:18:00 |   |  
  |   | 2023-05-22 06:18:00 | goroutine 201 [running]: |  
  |   | 2023-05-22 06:18:00 | github.com/prometheus/client_golang/prometheus.(*CounterVec).WithLabelValues(0xc00000f228, 0xc00010bf40, 0x4, 0x4, 0x7fe41e0f0398, 0xc0000f1560) |  
  |   | 2023-05-22 06:18:00 | /go/pkg/mod/github.com/prometheus/client_golang@v1.11.0/prometheus/counter.go:248 +0xda |  
  |   | 2023-05-22 06:18:00 | github.com/TykTechnologies/tyk-pump/pumps.(*PrometheusPump).WriteData(0xc00013f500, 0x1ddd6c0, 0xc00010b880, 0xc000447d40, 0x11, 0x11, 0x11, 0x11) |  
  |   | 2023-05-22 06:18:00 | /__w/tyk-pump/tyk-pump/pumps/prometheus.go:181 +0x705 |  
  |   | 2023-05-22 06:18:00 | main.execPumpWriting.func2(0xc0004af5c0, 0x1ddd6c0, 0xc00010b880, 0x1df5d00, 0xc00013f500, 0xc0005ca2c0) |  
  |   | 2023-05-22 06:18:00 | /__w/tyk-pump/tyk-pump/main.go:368 +0xaa |  
  |   | 2023-05-22 06:18:00 | created by main.execPumpWriting |  
  |   | 2023-05-22 06:18:00 | /__w/tyk-pump/tyk-pump/main.go:365 +0x3b0

```

**Comment by tbuchaillot** [comment]
Thanks for raising it, @patriziobrunops 🚀  - I added it to our backlog for prioritization.

**Comment by caroltyk** [comment]
It is fixed by #652 and will be available in the next release.

**Comment by oluwaseyeayinla** [comment]
Fixed in v1.8.2. Closing out

**Comment by olenotron** [comment]
> This might be root cause: [prometheus/client_golang#805](https://github.com/prometheus/client_golang/issues/805)
> 
> The `prometheus/golang_client` pump is using is very old. This issue was fixed over 2 years ago.

This issue was not fixed, and the version upgrade in https://github.com/TykTechnologies/tyk-pump/pull/652 didn't resolve it. In v.1.15.0, it is still there. This is what I got today:
```
  panic: label value "/\xc0" is not valid UTF-8                                                                                                                                                                                                                                                                                                                                                                                             
  prometheus.(*HistogramVec).WithLabelValues  client_golang@v1.16.0/prometheus/histogram.go:1125                                                                                                                                                                                                                                                                                                                                            
  pumps.(*PrometheusMetric).Observe           pumps/prometheus.go:528                                                                                                                                                                                                                                                                                                                                                                       
  pumps.(*PrometheusPump).processMetric        pumps/prometheus.go:319                                                                                                                                                                                                                                                                                                                                                                      
  pumps.(*PrometheusPump).WriteData            pumps/prometheus.go:344                                                                                                                                                                                                                                                                                                                                                                      
  created by main.execPumpWriting              main.go:456
```

You can reproduce it using `curl -sv 'https://<gw>/<listen-path>/%C0' -H '<auth>'` (with `track_all_paths: true`) →
pump panics on the next purge.

It makes the Prometheus pump panic and lose the in-flight batch from Redis. The batch is deleted from Redis before the panic, and the pumps next in line never get these records.
